src/main/java/wayoftime/bloodmagic/common/tile/TileSpectral.java

```diff
@@ -82,4 +86,23 @@ public CompoundTag serialize(CompoundTag tag)
 		tag.put("BlockState", NbtUtils.writeBlockState(storedBlock));
 		return tag;
 	}
+
+	public void tick()
+	{
+		BlockState state = this.getBlockState();
```

Comment:
The reason the blocks vanish so quickly is that the BE is ticked every tick. Instead, use a tick_counter and check against the DECAY_RATE and only then tick it.